sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/domain/GenIdEntity.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos.domain;
+
+import com.azure.spring.data.cosmos.core.mapping.GeneratedValue;
+import org.springframework.data.annotation.Id;
+
+import java.util.Objects;
+
+public class GenIdEntity {
+
+    @Id
+    @GeneratedValue
```

Comment:
what happens if a field of non String type is annotated with `@ GeneratedValue` do we support that? or do we error out?

Reply:
We error it out that `@GeneratedValue` can only be used on a string id.